src/guiapi/src/st7789v.cpp

```diff
@@ -535,11 +535,42 @@ void st7789v_draw_png_ex(uint16_t point_x, uint16_t point_y, FILE *pf, uint32_t
             { // these vars must be destroyed before label _e_1
                 uint16_t w = png_get_image_width(pp, ppi);
                 uint16_t h = png_get_image_height(pp, ppi);
+
                 int rowsize = png_get_rowbytes(pp, ppi);
+                int pixsize = rowsize / w;
+
+                //check image type (indexed or other color type)
+                png_byte colorType = png_get_color_type(pp, ppi);
+
+                switch (colorType) {
+                case PNG_COLOR_TYPE_GRAY:
+                    //transform grayscale image to rgb 24 color depth
+                    png_set_gray_to_rgb(pp);
+                    //pixel size is 3 bytes
+                    pixsize = 3;
+                    //check if alpha chanell is present if yes then add it and increase pixelSize
+                    if (png_get_valid(pp, ppi, PNG_INFO_tRNS)) {
+                        png_set_tRNS_to_alpha(pp);
+                        pixsize += 1;
+                    }
+                    break;
+                case PNG_COLOR_TYPE_PALETTE:
+                    //bit depth in palette is always 8 bits per sample (24 bits per color) so pixel size is 3 bytes
+                    pixsize = 3;
+                    png_set_palette_to_rgb(pp);
```

Comment:
nice :+1: - so the png lib already can expand the source image into RGB on its own, right?